src/resolver/mixins.cc

```diff
@@ -0,0 +1,353 @@
+          GlobalManifestRaw( std::move( registryRaw ) ) );
+      } );
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+argparse::Argument &
+EnvironmentMixin::addManifestFileOption( argparse::ArgumentParser & parser )
+{
+  return parser.add_argument( "--manifest" )
+    .help( "The path to the `manifest.{toml,yaml,json}' file." )
+    .metavar( "PATH" )
+    .action( [&]( const std::string & strPath )
+             { this->initManifestPath( nix::absPath( strPath ) ); } );
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+argparse::Argument &
+EnvironmentMixin::addManifestFileArg( argparse::ArgumentParser & parser,
+                                      bool                       required )
+{
+  argparse::Argument & arg
+    = parser.add_argument( "manifest" )
+        .help( "The path to the project's `manifest.{toml,yaml,json}' file." )
+        .metavar( "MANIFEST-PATH" )
+        .action( [&]( const std::string & strPath )
+                 { this->initManifestPath( nix::absPath( strPath ) ); } );
+  return required ? arg.required() : arg;
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+argparse::Argument &
+EnvironmentMixin::addLockfileOption( argparse::ArgumentParser & parser )
+{
+  return parser.add_argument( "--lockfile" )
+    .help( "The path to the projects existing `manifest.lock' file." )
+    .metavar( "PATH" )
+    .action( [&]( const std::string & strPath )
+             { this->initLockfilePath( nix::absPath( strPath ) ); } );
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+argparse::Argument &
+EnvironmentMixin::addFloxDirectoryOption( argparse::ArgumentParser & parser )
+{
+  return parser.add_argument( "--dir", "-d" )
+    .help( "The directory to search for `manifest.{json,yaml,toml}' and "
+           "`manifest.lock`." )
+    .metavar( "PATH" )
+    .nargs( 1 )
+    .action(
+      [&]( const std::string & strPath )
+      {
+        std::filesystem::path dir( nix::absPath( strPath ) );
+        /* Try to locate lockfile. */
+        auto path = dir / "manifest.lock";
+        if ( std::filesystem::exists( path ) )
+          {
+            this->initLockfilePath( path );
+          }
+        /* Locate manifest. */
+        if ( path = dir / "manifest.json"; std::filesystem::exists( path ) )
+          {
+            this->initManifestPath( path );
+          }
+        else if ( path = dir / "manifest.toml";
+                  std::filesystem::exists( path ) )
+          {
+            this->initManifestPath( path );
+          }
+        else if ( path = dir / "manifest.yaml";
+                  std::filesystem::exists( path ) )
+          {
+            this->initManifestPath( path );
+          }
+        else
+          {
+            throw EnvironmentMixinException(
+              "unable to locate a `manifest.{json,yaml,toml}' file "
```

Comment:
```suggestion
              "unable to locate a `manifest.toml' file "
```
This error message would be confusing for users to encounter since we only support TOML files

Reply:
Since `pkgdb` supports all 3 I'd prefer for the `flox` CLI to catch the error and print an alternative message.

Is that difficult to do on the Rust side?
If it is I can change it.

Reply:
In any case `--dir` isn't used on the `flox` CLI

Reply:
> Since `pkgdb` supports all 3 I'd prefer for the `flox` CLI to catch the error and print an alternative message.
> 
> Is that difficult to do on the Rust side? If it is I can change it.

It's not hard, we just currently print the entire error message from pkgdb.